### PR TITLE
Add a unit test which checks upstream media types

### PR DESCRIPTION
**Pull request by jrhee17**
Motivation:

Currently, human intervention is required to ensure that armeria is caught up for newly added media types.

1) Once in a while, someone has to check upstream to see if any media types were updated
2) Reviewers also need to check upstream implementation to ensure media types are forked correctly

I propose that we add a unit test `wellDefinedUpstreamMediaTypes` which checks if armeria is up-to-date with upstream `MediaType`s.

Modifications:

- Add a unit test `wellDefinedUpstreamMediaTypes`

Result:

- Slightly more automated `MediaType` checking

### CODE REVIEW

File: core/src/test/java/com/linecorp/armeria/common/MediaTypeTest.java [hunk 1]
```diff
@@ -85,24 +83,22 @@
 
     @Test // reflection
     public void testParse_useConstants() throws Exception {
-        for (MediaType constant : getConstants()) {
-            assertSame(constant, MediaType.parse(constant.toString()));
-        }
+        getConstants().forEach(constant -> {
```
Comment (trustin) [inline core/src/test/java/com/linecorp/armeria/common/MediaTypeTest.java:86]: This class is forked from Guava, so I'd suggest leaving as it is.

**Comment by jrhee17** [comment]
Failure has been simulated with the following commit:
https://github.com/line/armeria/runs/6806535578

**Review by ikhoon — APPROVED** [review]
Very nice!

**Review by minwoox — APPROVED** [review]
Thanks! 😄

**Comment by minwoox** [comment]
Thanks, @jrhee17! 😄

**Pull request merged**